litr/src/main/java/com/linkedin/android/litr/io/AudioRecordMediaSource.kt

```diff
@@ -0,0 +1,207 @@
+package com.linkedin.android.litr.io
+
+import android.media.*
+import android.os.Build
+import android.util.Log
+import androidx.annotation.RequiresApi
+import com.linkedin.android.litr.MimeType
+import com.linkedin.android.litr.exception.MediaSourceException
+import java.nio.ByteBuffer
+
+private const val TAG = "AudioRecordMediaSource"
+
+private const val MEDIA_FORMAT_PCM_KEY = "pcm-encoding"
+private const val AUDIO_FORMAT = AudioFormat.ENCODING_PCM_16BIT
+private const val BYTES_PER_SAMPLE = 2
+
+/**
+ * An implementation of MediaSource, which utilizes Android's {@link AudioRecord} to record audio
+ * from a given audio source (i.e. the device's microphone).
+ *
+ * This class requires Android M+ in order to support reading data from {@link AudioRecord} in a
+ * non-blocking way.
+ */
+@RequiresApi(Build.VERSION_CODES.M)
+class AudioRecordMediaSource(
+        private val audioSource: Int = DEFAULT_AUDIO_SOURCE,
+        private val sampleRate: Int = DEFAULT_SAMPLE_RATE,
+        private val isStereo: Boolean = false
+) : MediaSource {
+    private val channelCount: Int by lazy {
```

Comment:
Not sure how much are we saving by using `lazy` here. There is no complex class initialization going on. I think we can remove all `lazy` intiatizations here and below.

Reply:
Fair enough, I think I just got carried away :)

Reply:
Been there, done that... many times... :)